tornado/simple_httpclient.py

```diff
@@ -167,12 +167,16 @@ def fetch_impl(
     ) -> None:
         key = object()
         self.queue.append((key, request, callback))
-        if not len(self.active) < self.max_clients:
-            assert request.connect_timeout is not None
-            assert request.request_timeout is not None
+        assert request.connect_timeout is not None
+        assert request.request_timeout is not None
+        timeout = (
```

Comment:
It would be a (very slight) optimization to move this back under the `if len(self.active)` check (which is very rarely true):

```
if len(self.active) >= self.max_clients:
    timeout = ...
    if timeout:
        timeout_handle = add_timeout...
```

Reply:
Sounds good, done.

I left the asserts outside the if-block because I thought it might be better for them to apply consistently, rather than only after the queue fills (maybe causing confusing behavior change much later, or weirder exceptions in other contexts). I can put them inside the if block again also, it's not a strong opinion :)